src/agent/TransportService.ts

```diff
@@ -54,4 +57,7 @@ interface TransportSessionTable {
 
 export interface TransportSession {
   type: string
+  keys?: EnvelopeKeys
+  inboundMessage?: AgentMessage
```

Comment:
Why is `inboundMessage` optional? Because we also use it for outbound transport?

Reply:
Same for `keys`

Reply:
I think it is nice to make as much required as possible

Reply:
Sorry, I forgot to respond to this. The reason is that session is constructed outside the framework where `keys`, `connection`, or `inboundMessage` are not available. If these attributes are gathered at the same place, we can maybe introduce another data structure similar to `context` in `InboundMessageContext` and set it via a setter function.